admin/app/components/solidus_admin/ui/modal/component.html.erb

```diff
@@ -0,0 +1,33 @@
+<dialog data-controller=<%= stimulus_id %> id=<%= @id %> class="relative z-10" aria-labelledby="modal-title" aria-modal="true">
+  <div class="fixed inset-0 bg-gray-500 bg-opacity-75"></div>
+
+  <div class="fixed inset-0 z-10 overflow-y-auto">
+    <div class="flex min-h-full items-end justify-center p-4 text-center sm:items-center sm:p-0">
+      <div class="relative transform overflow-hidden rounded-lg bg-white text-left shadow-xl sm:my-8 sm:w-full sm:max-w-lg divide-y">
+        <div class="flex justify-between pb-4 pt-5 sm:p-6 sm:pb-4">
+          <div class="mt-3 text-center sm:mt-0 sm:text-left">
+            <h3 class="text-xl font-semibold leading-6 text-gray-900" id="modal-title">
+              <%= @title %>
+            </h3>
+          </div>
+          <button type="button" class="close" data-action=<%= "#{stimulus_id}#toggleModal" %> aria-label="Close" <%= "data-#{stimulus_id}-modal-id-param=#{@id}" %>>
+            <span aria-hidden="true">&times;</span>
+          </button>
```

Comment:
Why not use a `component('ui/button')` for this button?

Reply:
Unfortunately our button component only allows primary, secondary and ghost "layouts"; the `x` button on the modal does not match any of those.